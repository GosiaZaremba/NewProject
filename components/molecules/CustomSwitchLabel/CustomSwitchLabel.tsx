import React, { forwardRef, useImperativeHandle, useRef } from 'react';
import { CustomSwitch, CustomSwitchReference, Label } from '../../atoms';
import { View } from 'react-native';

export type Props = {
    trackColorFalse: string;
    trackColorTrue: string;
    thumbColorOn: string;
    thumbColorOff: string;
    labelText: string;
    testID?: string;
};

export type CustomSwitchLabelReference = {
    getValue: () => boolean;
};

export const CustomSwitchLabelWithReference: React.ForwardRefRenderFunction<
    CustomSwitchLabelReference,
    Props
> = (
    {
        trackColorFalse,
        trackColorTrue,
        thumbColorOn,
        thumbColorOff,
        labelText,
        testID,
    },
    ref
) => {
    const switchReference = useRef<CustomSwitchReference>(null);
    useImperativeHandle(ref, () => ({
        getValue: () => {
            return switchReference.current?.getValue() || false;
        },
    }));
    return (
        <View>
            <CustomSwitch
                trackColorFalse={trackColorFalse}
                trackColorTrue={trackColorTrue}
                thumbColorOn={thumbColorOn}
                thumbColorOff={thumbColorOff}
                ref={switchReference}
                testID={`${testID}-switch`}
            />
            <Label text={labelText} testID={`${testID}-label`} />
        </View>
    );
};

export const CustomSwitchLabel = forwardRef(CustomSwitchLabelWithReference);
