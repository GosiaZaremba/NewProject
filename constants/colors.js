export const Colors = {
  pink: '#fc296f',
  red: '#e63039',
  navy: '#39539b',
  blue: '#3872c3',
  grey: '#908D8E',
  light: '#F5F0F2',
};
