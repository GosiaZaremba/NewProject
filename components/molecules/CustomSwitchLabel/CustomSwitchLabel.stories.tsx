import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { CustomSwitchLabel } from './CustomSwitchLabel';
import { color, text } from '@storybook/addon-knobs';
import { Colors } from '../../../constants/colors';

storiesOf('molecules/CustomSwitchLabel', module)
    .addDecorator((story) => story())
    .add('default', () => (
        <CustomSwitchLabel
            trackColorFalse={color('trackColorFalse', Colors.pink)}
            trackColorTrue={color('trackColorTrue', Colors.pink)}
            thumbColorOn={color('thumbColorOn', Colors.pink)}
            thumbColorOff={color('thumbColorOff', Colors.pink)}
            labelText={text('Label Text', 'Label Text')}
        />
    ));
