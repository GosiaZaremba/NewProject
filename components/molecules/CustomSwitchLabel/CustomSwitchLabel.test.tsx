import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CustomSwitchLabel } from './CustomSwitchLabel';
import { Colors } from '../../../constants/colors';

describe('Molecule SwitchLabel', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <CustomSwitchLabel
                    trackColorFalse={Colors.grey}
                    trackColorTrue={Colors.pink}
                    thumbColorOn={Colors.grey}
                    thumbColorOff={Colors.grey}
                    labelText={'Molecule switch label'}
                    testID={'molecule-switchLabel'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <CustomSwitchLabel
                trackColorFalse={Colors.grey}
                trackColorTrue={Colors.pink}
                thumbColorOn={Colors.grey}
                thumbColorOff={Colors.grey}
                labelText={'Molecule switch label'}
                testID={'molecule-switchLabel'}
            />
        );
        const customSwitchLabel = getByTestId('molecule-switchLabel');
        // fireEvent.changeText(customInputLabel, '123');
        // expect(customInputLabel.props.value).toBe('123');
        // expect(customInputLabel.props.placeholder).toEqual('Custom InputLabel');
        // expect(customInputLabel.props.secureTextEntry).toEqual(false);
        // expect(customInput.props.onChangeText()).toHaveReturned();
        // console.log(customSwitchLabel.props);
    });
});
