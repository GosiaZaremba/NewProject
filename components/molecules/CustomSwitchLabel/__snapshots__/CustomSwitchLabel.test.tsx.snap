// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule SwitchLabel Component Snapshot 1`] = `
<View
  testID="molecule-switchLabel"
>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#fc296f"
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    thumbTintColor="#908D8E"
    tintColor="#908D8E"
    value={false}
  />
  <Text
    style={
      Object {
        "color": "black",
        "marginBottom": 2,
      }
    }
  >
    Molecule switch label
  </Text>
</View>
`;
